### PR TITLE
Day 4 HW completed

### PR DESCRIPTION
**Pull request by rohitraghavan**

### CODE REVIEW

File: HW04_ch08_ex05.py [hunk 1]
```diff
@@ -2,3 +2,31 @@
...
+		#Convert char to number
+		numchar = ord(char)
+		#Rotate the chars
+		newnumchar = numchar + rotationno
+		#Handle upper case chars which go outside bounds on rotating
+		if 65 <= numchar <= 90 and newnumchar > 90:
+			newnumchar = 65 + (newnumchar - 90 -1)
+		#Handle lower case chars which go outside bounds on rotating
+		elif 97 <= numchar <= 122 and newnumchar > 122:
+			newnumchar = 97 + (newnumchar - 122 -1)
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: add space: `newnumchar = 97 + (newnumchar - 122 - 1)`

File: D04_ex00.py [hunk 2]
```diff
@@ -11,18 +11,36 @@
...
 ################################################################################
 # Imports
-
+import random
 
 # Body
-
+def guess():
+	#Set a random number
+	randomno = random.randrange(1, 25, 2)
```
Comment (danielsgriffin) [inline D04_ex00.py]: Please run this code and see what the actual output is. Something is fishy.

File: HW04_ch07_ex02.py [hunk 3]
```diff
@@ -21,15 +21,25 @@
 
 ###############################################################################
 # Imports
-
+import math
 
 # Body
+def eval_loop():
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: Instructions: "return the value of the last expression it evaluated"

**Comment by danielsgriffin** [comment]
Good job. Some instructions seem to be missed.

File: HW04_ch08_ex05.py [hunk 4]
```diff
@@ -2,3 +2,31 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
 # function.
+
+def rotate(word, rotationno):
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: rotate_word

File: HW04_ch08_ex05.py [hunk 5]
```diff
@@ -2,3 +2,31 @@
 # See Chapter 8: Strings Exercise 5 for the task.
 # Please do provide function calls that test/demonstrate your
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]:
```
Book Examples:
 OK  'A'     by 3   => got: 'D'     expected: 'D'
 OK  'Z'     by 1   => got: 'A'     expected: 'A'
 OK  'cheer' by 7   => got: 'jolly' expected: 'jolly'
  X  'melon' by -10 => got: 'c[bed' expected: 'cubed'
 OK  'IBM'   by -1  => got: 'HAL'   expected: 'HAL'

Daniel's Tests:
 OK  'abc'   by 1   => got: 'bcd'   expected: 'bcd'
 OK  'xyz'   by 1   => got: 'yza'   expected: 'yza'
 OK  'ABC'   by 1   => got: 'BCD'   expected: 'BCD'
 OK  'XYZ'   by 1   => got: 'YZA'   expected: 'YZA'
 OK  'Az'    by 1   => got: 'Ba'    expected: 'Ba'
 OK  'Aa'    by 26  => got: 'Aa'    expected: 'Aa'
  X  'abc'   by -1  => got: '`ab'   expected: 'zab'
 OK  'xyz'   by -1  => got: 'wxy'   expected: 'wxy'
  X  'ABC'   by -1  => got: '@AB'   expected: 'ZAB'
 OK  'XYZ'   by -1  => got: 'WXY'   expected: 'WXY'
  X  'Az'    by -1  => got: '@y'    expected: 'Zy'
  X  'Aa'    by -26 => got: "'G"    expected: 'Aa'
```

File: D04_ex00.py (hunk 2 above)
Comment (niavivek) [inline D04_ex00.py]: This generates only odd numbers --> randrange(start,stop,step).

File: HW04_ch08_ex05.py [hunk 6]
```diff
@@ -2,3 +2,31 @@
...
+
+def rotate(word, rotationno):
+	newword = ""
+	for char in word:
+		#Convert char to number
+		numchar = ord(char)
+		#Rotate the chars
+		newnumchar = numchar + rotationno
+		#Handle upper case chars which go outside bounds on rotating
+		if 65 <= numchar <= 90 and newnumchar > 90:
```
Comment (niavivek) [inline HW04_ch08_ex05.py]: Doesn't work for negative rotation. You only have a case for rotating in the +ve direction. Think about 'A' rotated in -ve direction, the ASCII value will be out of range or alphabets.

**Pull request closed without merging**